riscv/src/register/mstatus.rs

```diff
@@ -81,46 +96,106 @@ impl Mstatus {
     /// Supervisor Interrupt Enable
     #[inline]
     pub fn sie(&self) -> bool {
-        self.bits & (1 << 1) != 0
+        self.bf_extract(1, 1) != 0
+    }
+
+    /// Update Supervisor Interrupt Enable
+    ///
+    /// Note this updates a previously read [`Mstatus`] value, but does not
+    /// affect the mstatus CSR itself. See [`set_sie`]/[`clear_sie`] to directly
+    /// update the CSR.
+    #[inline]
+    pub fn set_sie(&self, sie: bool) -> Self {
```

Comment:
```suggestion
    pub fn set_sie(&mut self, sie: bool) {
```

Reply:
From [`cortex-m`](https://github.com/rust-embedded/cortex-m/blob/4395baee03d57a4e1763c427073da5f33ec3b17e/cortex-m/src/register/fpscr.rs#L32):
```rust
#[inline]
pub fn set_n(&mut self, n: bool) {
    let mask = 1 << 31;
    match n {
        true => self.bits |= mask,
        false => self.bits &= !mask,
    }
}
```
So let's stick with the classic setter approach with mutable reference and no return value.

Reply:
What about returning `&mut Self` so you can chain them?

Reply:
I like your idea!